src/Security/RolesMatrixBuilder.php

```diff
@@ -0,0 +1,299 @@
+                'read_only' => !$this->authorizationChecker->isGranted($name),
+            ];
+
+            foreach ($rolesHierarchy as $role) {
+                if (false === array_key_exists($role, $this->rolesHierarchy)
+                    && !isset($roles[$role])
+                    && false === $this->recursiveArraySearch($role, $roles)
+                ) {
+                    $roles[$role] = [
+                        'read_only' => !$this->authorizationChecker->isGranted($role),
+                    ];
+                }
+            }
+        }
+
+        return $roles;
+    }
+
+    public function getExclude(): array
+    {
+        return $this->exclude;
+    }
+
+    public function addExclude(string $exclude): void
+    {
+        $this->exclude[] = $exclude;
+    }
+
+    public function getLabelPermission(): array
+    {
+        return $this->labelPermission;
+    }
+
+    public function getLabelAdmin(): array
+    {
+        return $this->labelAdmin;
+    }
+
+    public function getAdminRolesForView(): array
+    {
+        $roles = [];
+        // get roles from the Admin classes
+        foreach ($this->pool->getAdminServiceIds() as $id) {
+            try {
+                $admin = $this->pool->getInstance($id);
+            } catch (\Exception $e) {
+                continue;
+            }
+
+            if (in_array($id, $this->exclude)) {
+                continue;
+            }
+
+            $isMaster = ($admin->isGranted('MASTER') || $admin->isGranted('OPERATOR') ||
+                $this->authorizationChecker->isGranted($this->pool->getOption('role_super_admin')));
+
+            $securityHandler = $admin->getSecurityHandler();
+            // TODO get the base role from the admin or security handler
+            $baseRole = $securityHandler->getBaseRole($admin);
+            $groupPermission = $admin->getSecurityInformation();
+            $this->labelPermission = array_keys($groupPermission);
+            $this->labelAdmin[] = $admin->getTranslator()->trans($admin->getLabel());
+
+            if (0 == strlen($baseRole)) { // the security handler related to the admin does not provide a valid string
+                continue;
+            }
+
+            $roles[$baseRole] = [
+                'label' => $admin->getTranslator()->trans($admin->getLabel()),
+            ];
+
+            foreach ($groupPermission as $name => $item) {
+                $role = sprintf($baseRole, $name);
+                $roles[$baseRole]['permissions'][$name] =
+                    !$isMaster && !$this->authorizationChecker->isGranted($role);
+            }
+        }
+
+        return $roles;
+    }
+
+    private function iterateAdminRoles(callable $func): void
+    {
+        // get roles from the Admin classes
+        foreach ($this->pool->getAdminServiceIds() as $id) {
+            try {
+                $admin = $this->pool->getInstance($id);
+            } catch (\Exception $e) {
+                continue;
```

Comment:
Same here

Reply:
You should pass the exception as logging parameter `exception`
https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-3-logger-interface.md#13-context